External/FEXCore/Source/Interface/Core/ArchHelpers/Relocations.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+#include <FEXCore/IR/IR.h>
+#include <vector>
+
+namespace FEXCore::CPU {
+  enum class RelocationTypes : uint8_t {
+    // 8 byte literal in memory for symbol
+    // Aligned to struct RelocNamedSymbolLiteral
+    RELOC_NAMED_SYMBOL_LITERAL,
+
+    // Fixed size named thunk move
+    // 4 instruction constant generation on AArch64
+    // 64-bit mov on x86-64
+    // Aligned to struct RelocNamedThunkMove
+    RELOC_NAMED_THUNK_MOVE,
+
+    // Fixed size guest RIP move
+    // 4 instruction constant generation on AArch64
+    // 64-bit mov on x86-64
+    // Aligned to struct RelocGuestRIPMove
+    RELOC_GUEST_RIP_MOVE,
+  };
+
+  struct RelocationTypeHeader final {
+    RelocationTypes Type;
+  };
+
+  struct RelocNamedSymbolLiteral final {
+    enum class NamedSymbol : uint8_t {
+      ///< Thread specific relocations
+      // JIT Literal pointers
+      SYMBOL_LITERAL_EXITFUNCTION_LINKER,
+    };
+
+    RelocationTypeHeader Header{};
+
+    // Offset in to the code section to begin the relocation
+    uint64_t Offset{};
+
+    NamedSymbol Symbol;
+  };
+
+  struct RelocNamedThunkMove final {
+    RelocationTypeHeader Header{};
+
+    // The thunk SHA256 hash
+    IR::SHA256Sum Symbol;
+
+    // Offset in to the code section to begin the relocation
+    uint64_t Offset{};
+
+    // GPR index the constant is being moved to
+    uint8_t RegisterIndex;
```

Comment:
Moving this member in front of `Symbol` allows this struct to fit into 32 bytes of member again, doesn't it?

Reply:
Moved the uint8_t RegisterIndex in the two relocations to be just after the header definitions so it packs better.
Resolved.